### ISSUE TITLE
[BUG REPORT] 1.wav contains NaN

### ISSUE THREAD

**Issue by kamileqq** [gitmylo/audio-webui#90]
When i'm clicking extract pitches it shows that "1.wav contains NaN" and like this for 59 other files, these files have sound in them.
I tried doing it on pm and harvest extraction methods

**Expected behavior**
webui should extract pitches succesfully

**Screenshots**
![image](https://github.com/gitmylo/audio-webui/assets/58258293/4e13e77a-4462-4c25-8b09-12f4380c539a)
![image](https://github.com/gitmylo/audio-webui/assets/58258293/c09ac48a-7b1b-4afd-9c66-ed1cee80797b)

**Comment by gitmylo** [comment]
ffmpeg the sounds back to wav. Also don't use pm for training, use harvest or any crepe variant

**Comment by kamileqq** [comment]
hello now it started throwing "Exception: Each element of kernel_size should be odd. Skipping" many times
![image](https://github.com/gitmylo/audio-webui/assets/58258293/eee314f7-c3b9-4fef-9596-1013dfe052ac)
![image](https://github.com/gitmylo/audio-webui/assets/58258293/37cff847-ea98-49ac-bee3-da95aefa0836)

**Comment by gitmylo** [comment]
that might be an issue with your filter radius

**Comment by kamileqq** [comment]
yeah setting the filter radius to 0 or 1 fixed the "kernel size should be odd" issue but i still get the NaN

**Comment by galagaygay** [comment]
I also have this problem, no matter what is set in the train interface, an error is reported. (NaN)

windows10 conda Python3.8.10

**Comment by gitmylo** [comment]
![image](https://github.com/gitmylo/audio-webui/assets/36931363/8e80b2ab-83d4-431e-b2f4-79b2e59395b8)
I can't really reproduce it, do you have ffmpeg installed, and can you check the files in the `data/training/RVC/Test/0_16k/` folder? these should be playable wav files, if not, something went wrong earlier down the line.

also check if you have ffmpeg installed, that could also cause this

**Comment by kamileqq** [comment]
![image](https://github.com/gitmylo/audio-webui/assets/58258293/a214dba3-a557-4b4c-af42-198d306a2a6d)
![image](https://github.com/gitmylo/audio-webui/assets/58258293/dd142b57-2337-4145-b499-2e3984addd1d)

**Comment by gitmylo** [comment]
There is no debug mode or CLI mode. The error is printed here:
 https://github.com/gitmylo/audio-webui/blob/559bd7ebfab0162a353e585188a3bae9d309320b/webui/ui/tabs/training/training/rvc_workspace.py#L302
Which implies at least one of the features contains NaN.

You could try a different dataset, see if that makes a difference.

**Comment by diogoalmeida1991** [comment]
I changed the files into WAV 16000 in Audacity, but the problem persist.

The line wich create the bug is "logits = model.extract_features(**inputs)".

Follow bellow the variables before and after this line:

model variable:

HubertModel(
  (feature_extractor): ConvFeatureExtractionModel(
    (conv_layers): ModuleList(
      (0): Sequential(
        (0): Conv1d(1, 512, kernel_size=(10,), stride=(5,), bias=False)
        (1): Dropout(p=0.0, inplace=False)
        (2): Fp32GroupNorm(512, 512, eps=1e-05, affine=True)
        (3): GELU(approximate='none')
      )
      (1-4): 4 x Sequential(
        (0): Conv1d(512, 512, kernel_size=(3,), stride=(2,), bias=False)
        (1): Dropout(p=0.0, inplace=False)
        (2): GELU(approximate='none')
      )
      (5-6): 2 x Sequential(
        (0): Conv1d(512, 512, kernel_size=(2,), stride=(2,), bias=False)
        (1): Dropout(p=0.0, inplace=False)
        (2): GELU(approximate='none')
      )
    )
  )
  (post_extract_proj): Linear(in_features=512, out_features=768, bias=True)
  (dropout_input): Dropout(p=0.1, inplace=False)
  (dropout_features): Dropout(p=0.1, inplace=False)
  (encoder): TransformerEncoder(
    (pos_conv): Sequential(
      (0): Conv1d(768, 768, kernel_size=(128,), stride=(1,), padding=(64,), groups=16)
      (1): SamePad()
      (2): GELU(approximate='none')
    )
    (layers): ModuleList(
      (0-11): 12 x TransformerSentenceEncoderLayer(
        (self_attn): MultiheadAttention(
          (dropout_module): FairseqDropout()
          (k_proj): Linear(in_features=768, out_features=768, bias=True)
          (v_proj): Linear(in_features=768, out_features=768, bias=True)
          (q_proj): Linear(in_features=768, out_features=768, bias=True)
          (out_proj): Linear(in_features=768, out_features=768, bias=True)
        )
        (dropout1): Dropout(p=0.1, inplace=False)
        (dropout2): Dropout(p=0.0, inplace=False)
        (dropout3): Dropout(p=0.1, inplace=False)
        (self_attn_layer_norm): LayerNorm((768,), eps=1e-05, elementwise_affine=True)
        (fc1): Linear(in_features=768, out_features=3072, bias=True)
        (fc2): Linear(in_features=3072, out_features=768, bias=True)
        (final_layer_norm): LayerNorm((768,), eps=1e-05, elementwise_affine=True)
      )
    )
    (layer_norm): LayerNorm((768,), eps=1e-05, elementwise_affine=True)
  )
  (layer_norm): LayerNorm((512,), eps=1e-05, elementwise_affine=True)
  (final_proj): Linear(in_features=768, out_features=256, bias=True)
)

Variable input:
{'source': tensor([[ 1.3542e-04,  3.9577e-04,  8.7976e-05,  ..., -2.2034e-02,
         -3.8208e-02, -4.8920e-02]], device='cuda:0', dtype=torch.float16), 'padding_mask': tensor([[False, False, False,  ..., False, False, False]], device='cuda:0'), 'output_layer': 12}
		 
Variable logits:
(tensor([[[nan, nan, nan,  ..., nan, nan, nan],
         [nan, nan, nan,  ..., nan, nan, nan],
         [nan, nan, nan,  ..., nan, nan, nan],
         ...,
         [nan, nan, nan,  ..., nan, nan, nan],
         [nan, nan, nan,  ..., nan, nan, nan],
         [nan, nan, nan,  ..., nan, nan, nan]]], device='cuda:0',
       dtype=torch.float16), tensor([[False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False]], device='cuda:0'))

**Comment by gitmylo** [comment]
I'm still not sure if it's a hardware/software difference, or something with the audio files.

**Comment by diogoalmeida1991** [comment]
I find the bug! Change in "def pitch_extract():" the line 263 "model = model.half()" to "model = model.float()" and the line 285 ""source": features.half().to(device)" to ""source": features.float().to(device)". This occur due 16xx gpu.

Source: https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/issues/204

**Comment by gitmylo** [comment]
> I find the bug! Change in "def pitch_extract():" the line 263 "model = model.half()" to "model = model.float()" and the line 285 ""source": features.half().to(device)" to ""source": features.float().to(device)". This occur due 16xx gpu.
> 
> Source: [RVC-Project/Retrieval-based-Voice-Conversion-WebUI#204](https://github.com/RVC-Project/Retrieval-based-Voice-Conversion-WebUI/issues/204)

Thank you so much, I'll fix this immediately.

**Comment by gitmylo** [comment]
It will use a bit more memory, but training uses more than it anyways, so that won't be a problem.
